cranelift/frontend/src/frontend.rs

```diff
@@ -968,6 +968,7 @@ mod tests {
         sample_function(true)
     }
 
+    #[cfg(target_arch = "x86_64")]
```

Comment:
As bjorn3 pointed out, you might need something like `#[cfg_attr(feature = "x86", ignore)]` here and below instead.

Reply:
I think you are missing a `not()`.

Reply:
Thank you! However, `x86` is a feature of `cranelift-codegen`, not `cranelift-frontend`. Is there a way to check from the _frontend_ crate if the x86 backend is available?
(sorry if it's a newbie question!)

Reply:
Ah, it's a very good question. This particular crate shouldn't have tests taht depend on specific features of other crates, otherwise they wouldn't be run by our automation.

The behavior of `cranelift-codegen`, in the absence of specifying a target architecture for which one would like to compile, is to target the host architecture (this happens in `cranelift-codegen`'s `build.rs` script). But that doesn't _enable_ the Cargo feature either, so we couldn't even guard against the `cranelift-codegen`'s `x86` feature here.

So I think that checking against the `target_arch` is actually the right way to go: on a x86_64 host, the x86_64 target will be enabled by default. Sorry for the back and forth here, you were probably right in the first place :)
Would still be nice to use the `cfg_attr` directive instead.

Reply:
Thanks, I've updated the PR accordingly.